### ISSUE TITLE
copy and paste does not work

### ISSUE THREAD

**Issue by vks** [integer32llc/rust-playground#96]
It seems like it is not possible to copy and/or paste text, neither via keyboard shortcuts or the context menu. (Firefox 50.1.0, Ubuntu)

**Comment by shepmaster** [comment]
It's certainly possible to do this; in fact it's one of my most frequent usages, and it does work in Firefox 50.1.0 on macOS. 

I think we will need more details on how to reproduce this. Which piece of text are you unable to copy? I'm assuming it's the primary code chunk? If so, could you please try switching to the simple editor and copying the text there? Could you also try copying the text "channel" above the `nightly` / `stable` selectors?

![image](https://cloud.githubusercontent.com/assets/174509/22208345/8cd5331e-e150-11e6-8ead-6500083e0231.png)

**Comment by shepmaster** [comment]
Closing for the best reason ever: It Works On My Machine 😇

**Comment by vks** [comment]
Sorry for not replying. I think it might indeed be a problem on my side, so closing this seems like the right thing to do.

**Comment by shepmaster** [comment]
This may actually be ajaxorg/ace#2794, it looks highly relevant.

**Comment by vks** [comment]
I still can't copy and paste from the primary code, neither with keybindings or the context menu. Copying from other parts of the website (like "channel") works.

Switching to the simple editor fixes the problems, so it might indeed be the issue you linked.

**Comment by bopoh13** [comment]
You can choose the editor. How with [other](https://github.com/codemirror/CodeMirror/issues/4589) editors?

**Comment by shepmaster** [comment]
From [duplicate issue](https://github.com/integer32llc/rust-playground/issues/227#issuecomment-341412164)

> I'm not sure what you mean. I still care about it. What do you expect me to do? It seems to be the bug in Ace you linked, so my plan was to wait until it is fixed upstream. (I guess I don't care enough to invest the time to fix it myself. Especially since the simple editor works.)

I apologize, I didn't intend any slight. 

What I was attempting to convey is that over the course of a year, only two people have mentioned the issue. This isn't something that many people experience and frankly it can't be fixed except by someone who can reproduce it (or at least a lot of assistance from them); there's nothing *I* can do to fix something I can't see.

**Comment by dhardy** [comment]
I would suggest you leave it open for now just so people can find the issue.

**Comment by shepmaster** [comment]
> I would suggest you leave it open for now 

Yep, that's my plan. After a month or two of inactivity, I'll close it again, because I can't reproduce it much less fix it. Hopefully someone who experiences the problem fixes it in that timeframe!

> just so people can find the issue.

The GitHub search and Google search both work equally well for closed as well as open issues.

**Comment by vks** [comment]
This issue persists after upgrading to Ubuntu 16.04 with Firefox 57.0.

**Comment by vks** [comment]
As mentioned in ajaxorg/ace#2794, this happens when clipboard events are disabled. Enabling them fixes the issue for me.

I suggest to add a troubleshooting section to the README with an entry about this and to close the issue.

**Comment by shepmaster** [comment]
> when clipboard events are disabled

@vks any idea how your clipboard events got disabled?

**Comment by vks** [comment]
Not sure, I don't remember disabling them intentionally.
